EventFilter/EcalRawToDigi/plugins/alpaka/DeclsForKernels.h

```diff
@@ -0,0 +1,43 @@
+#ifndef EventFilter_EcalRawToDigi_plugins_alpaka_DeclsForKernels_h
+#define EventFilter_EcalRawToDigi_plugins_alpaka_DeclsForKernels_h
+
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE::ecal::raw {
+
+  struct InputDataHost {
+    // delete the default constructor because alpaka buffers do not have a default constructor
+    InputDataHost() = delete;
+
+    explicit InputDataHost(const Queue& queue, size_t size, size_t nFeds)
+        : data{cms::alpakatools::make_host_buffer<unsigned char[]>(queue, size)},
+          offsets{cms::alpakatools::make_host_buffer<uint32_t[]>(queue, nFeds)},
+          feds{cms::alpakatools::make_host_buffer<int[]>(queue, nFeds)} {};
+
+    cms::alpakatools::host_buffer<unsigned char[]> data;
+    cms::alpakatools::host_buffer<uint32_t[]> offsets;
+    cms::alpakatools::host_buffer<int[]> feds;
+  };
```

Comment:
In principle, `InputDataHost` could be moved out of `ALPAKA_ACCELERATOR_NAMESPACE` into `namespace ecal::raw`, changing the constructor to 
```c++
    template <typename TQueue>
    explicit InputDataHost(const TQueue& queue, size_t size, size_t nFeds)
```
to make it work with any kind of queue.

Reply:
@makortel do you have a preference ?

Reply:
Hmm. I don't have a clear preference.

On one hand, moving the `InputDataHost` outside of `ALPAKA_ACCELERATOR_NAMESPACE` and templating the constructor over `TQueue` would be more consistent with the requirements for ED or ES host-side data products.

On the other hand, the `InputDataHost` seems like an implementation detail of `EcalRawToDigiPortable` EDProducer, and the queue behavior is pretty much tied to the backend (or `ALPAKA_ACCELERATOR_NAMESPACE`) in question.

If any wider use of `InputDataHost`, or use outside of `plugins/alpaka` subdirectory, would be foreseen, templating would make sense. Maybe the templating the constructor would lead to shorter symbol names?

Reply:
After re-reading the PR, I agree that it's easier to leave them as they are, since `InputDataHost` and `InputDataDevice` are only used internally to `EventFilter/EcalRawToDigi/plugins/alpaka/`.